Sources/LLVMSwift/TargetData.swift

```diff
@@ -168,6 +44,139 @@ public struct TargetData {
   public func sizeOfTypeInBits(_ type: IRType) -> Int {
     return Int(LLVMSizeOfTypeInBits(llvm, type.asLLVM()))
   }
+
+  /// Computes the number of bits necessary to hold a value of the given type
+  /// for this target environment.
+  ///
+  /// - parameter type: The type to compute the size of.
+  ///
+  /// - returns: The size of the type in bits.
+  public func abiSizeOfTypeInBits(_ type: IRType) -> Int {
+    return Int(LLVMSizeOfTypeInBits(llvm, type.asLLVM()))
+  }
+
+  /// The current platform byte order, either big or little endian.
+  public var byteOrder: ByteOrder {
+    return ByteOrder(llvm: LLVMByteOrder(llvm))
+  }
+
+  /// Creates a string representation of the target data.
+  public var layoutString: String {
+    let str = LLVMCopyStringRepOfTargetData(llvm)!
+    defer { free(str) }
+    return String(cString: str)
+  }
+
+  /// The integer type that is the same size as a pointer on this target.
+  /// This is analoguous to the `intptr_t` type in C++.
+  /// - parameters:
+  ///   - context: The context in which to derive the type (optional).
+  ///   - addressSpace: The address space in which to derive the type.
+  /// - returns: An IntegerType that is the same size as the pointer type
+  ///            on this target.
+  public func intPointerType(context: Context? = nil, addressSpace: Int? = nil) -> IntType {
+    let type: LLVMTypeRef
+    switch (context, addressSpace) {
+    case let (context?, addressSpace?):
+      type = LLVMIntPtrTypeForASInContext(context.llvm, llvm, UInt32(addressSpace))
+    case let (nil, addressSpace?):
+      type = LLVMIntPtrTypeForAS(llvm, UInt32(addressSpace))
+    case let (context?, nil):
+      type = LLVMIntPtrTypeInContext(context.llvm, llvm)
+    case (nil, nil):
+      type = LLVMIntPtrType(llvm)
+    }
+    return convertType(type) as! IntType // Guaranteed to succeed
+  }
+
+  /// Computes the preferred alignment of the given global for this target
+  ///
+  /// - parameter global: The global variable
+  /// - returns: The variable's preferred alignment in this target
+  public func preferredAlignment(of global: Global) -> Int {
+    return Int(LLVMPreferredAlignmentOfGlobal(llvm, global.asLLVM()))
+  }
+
+  /// Computes the preferred alignment of the given type for this target
+  ///
+  /// - parameter type: The type for which you're computing the alignment
+  /// - returns: The type's preferred alignment in this target
+  public func preferredAlignment(of type: IRType) -> Int {
+    return Int(LLVMPreferredAlignmentOfType(llvm, type.asLLVM()))
+  }
+
+  public func abiAlignment(of type: IRType) -> Int {
+    return Int(LLVMABIAlignmentOfType(llvm, type.asLLVM()))
+  }
+
+  public func callFrameAlignment(of type: IRType) -> Int {
```

Comment:
This seems like a wrapper for the API above that just happens to use function types.  Docs should be the same.